src/modules/px4iofirmware/mixer.cpp

```diff
@@ -287,19 +295,29 @@ mixer_tick(void)
 		/* mix */
 		mixed = mixer_mix_threadsafe(&outputs[0], &r_mixer_limits);
 
-		/* the pwm limit call takes care of out of band errors */
-		pwm_limit_calc(should_arm, should_arm_nothrottle, mixed, r_setup_pwm_reverse, r_page_servo_disarmed,
-			       r_page_servo_control_min, r_page_servo_control_max, outputs, r_page_servos, &pwm_limit);
+		/* if mixer_mix_threadsafe returns zero, it did nothing */
+		if (mixed != 0) {
```

Comment:
It looked like a nasty bug to me (forcing all outputs to zero because something is locked?), but it's quite possible I don't understand how/why the code works.